cmd/kubeadm/app/cmd/join.go

```diff
@@ -419,6 +421,20 @@ func (j *joinData) Run() error {
 		fmt.Printf("[join] Running pre-flight checks before initializing the new control plane instance\n")
 		preflight.RunInitMasterChecks(utilsexec.New(), initCfg, j.ignorePreflightErrors)
 
+		fmt.Printf("[join] Pulling control-plane images\n")
+		externalcfg := &kubeadmapiv1beta1.InitConfiguration{}
+		kubeadmscheme.Scheme.Default(externalcfg)
+		var cfgPath string
+		internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, externalcfg)
+		kubeadmutil.CheckErr(err)
+		containerRuntime, err := utilruntime.NewContainerRuntime(utilsexec.New(), internalcfg.GetCRISocket())
```

Comment:
these lines can be omitted:
```
		externalcfg := &kubeadmapiv1beta1.InitConfiguration{}
		kubeadmscheme.Scheme.Default(externalcfg)
		var cfgPath string
		internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, externalcfg)
		kubeadmutil.CheckErr(err)
```

change this to:
```
containerRuntime, err := utilruntime.NewContainerRuntime(utilsexec.New(), j.cfg.GetCRISocket())
```

the `j` object already has the `cfg` which is of type `JoinConfiguration` and it has a `GetCRISocket()` method...